src/test/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/group/TestRabbitMessageGroupBatchRouter.kt

```diff
@@ -0,0 +1,274 @@
+    private val connectionManager: ConnectionManager = mock {
+        on { configuration }.thenReturn(connectionConfiguration)
+        on { basicConsume(any(), any(), any()) }.thenReturn(monitor)
+    }
+
+    @Nested
+    inner class Publishing {
+        private val router = createRouter(mapOf(
+            "test-pine" to QueueConfiguration(
+                routingKey = "",
+                queue = "subscribe",
+                exchange = "test-exchange",
+                attributes = listOf("subscribe")
+            ),
+            "test-pin1" to QueueConfiguration(
+                routingKey = "test",
+                queue = "",
+                exchange = "test-exchange",
+                attributes = listOf("publish"),
+                filters = listOf(
+                    MqRouterFilterConfiguration(
+                        metadata = listOf(
+                            FieldFilterConfiguration(
+                                fieldName = "message_type",
+                                expectedValue = "test-message",
+                                operation = FieldFilterOperation.EQUAL
+                            )
+                        )
+                    )
+                )
+            ),
+            "test-pin2" to QueueConfiguration(
+                routingKey = "test2",
+                queue = "",
+                exchange = "test-exchange",
+                attributes = listOf("publish", "test"),
+                filters = listOf(
+                    MqRouterFilterConfiguration(
+                        metadata = listOf(
+                            FieldFilterConfiguration(
+                                fieldName = "message_type",
+                                expectedValue = "test-message",
+                                operation = FieldFilterOperation.EQUAL
+                            )
+                        )
+                    )
+                )
+            )
+        ))
+
+        @Test
+        fun `does not publish anything if all messages are filtered`() {
+            router.send(
+                MessageGroupBatch.newBuilder()
+                    .addGroups(MessageGroup.newBuilder()
+                        .apply { this += message("test-message1", Direction.FIRST, "test-alias") }
+                    ).build()
+            )
+
+            verify(connectionManager, never()).basicPublish(any(), any(), anyOrNull(), any())
+        }
+
+        @Test
+        fun `publishes to the correct pin according to attributes`() {
+            val batch = MessageGroupBatch.newBuilder()
+                .addGroups(MessageGroup.newBuilder()
+                    .apply { this += message("test-message", Direction.FIRST, "test-alias") }
+                ).build()
+            router.send(batch, "test")
+
+            val captor = argumentCaptor<ByteArray>()
+            verify(connectionManager).basicPublish(eq("test-exchange"), eq("test2"), anyOrNull(), captor.capture())
```

Comment:
```suggestion
            verify(connectionManager, times(1)).basicPublish(eq("test-exchange"), eq("test2"), anyOrNull(), captor.capture())
```

Reply:
It adds the 1 implicitly. No needs to do it manually